### PR TITLE
Support slash commands

### PR DESCRIPTION
**Pull request by DRSchlaubi**
Ported commands:

- [x] Channel
- [x] Help
- [x] Invite
- [x] Locale
- [x] Log
- [x] Prefix
- [x] Reputation (Splitted into Reputation and TopReputation)
- [x] RepSesstings
- [x] Roles
- [x] Scan
- [x] Scource
- [x] Thankwords

QA Checklist:
- [x] ephemeral is used on all errors messages

### CODE REVIEW

**Review by RainbowDashLabs — APPROVED** [review]

File: src/main/java/de/chojo/repbot/ReputationBot.java [hunk 1]
```diff
@@ -116,6 +117,7 @@ private void initBot() {
         var stateListener = new StateListener(dataSource);
         cleaner.scheduleAtFixedRate(stateListener, 0, 12, TimeUnit.HOURS);
 
+        // TODO: 28.05.2021 NE MENGE!
```
Comment (dlsf) [inline src/main/java/de/chojo/repbot/ReputationBot.java:120]: Self-explanatory

File: src/main/java/de/chojo/repbot/commands/Thankwords.java [hunk 2]
```diff
@@ -126,17 +173,87 @@ private boolean remove(MessageEventWrapper eventWrapper, CommandContext context)
...
+        var result = MessageAnalyzer.processMessage(guildSettings.getThankwordPattern(), message, guildSettings.getMaxMessageAge(), true, 0.85, 3);
+        var builder = new LocalizedEmbedBuilder(this.loc, event.getGuild());
+        if (result.getReceivers().isEmpty()) {
+            event.reply(loc.localize("command.thankwords.sub.check.match.noMatch")).queue();
+            return true;
+        }
+
+        var processResult = processMessage(event.getGuild(), result, builder);
+        if (processResult != null) {
+            event.reply(wrap(processResult)).queue();
```
Comment (dlsf) [inline src/main/java/de/chojo/repbot/commands/Thankwords.java:253]: I'm not really sure, but there might be a missing return statement

File: src/main/java/de/chojo/repbot/commands/Thankwords.java [hunk 3]
```diff
@@ -153,44 +270,53 @@ private boolean check(MessageEventWrapper eventWrapper) {
             return true;
         }
 
+        var processResult = processMessage(eventWrapper.getGuild(), result, builder);
+        if (processResult != null) {
+            eventWrapper.reply(processResult).queue();
```
Comment (dlsf) [inline src/main/java/de/chojo/repbot/commands/Thankwords.java:275]: Same here

**Review by dlsf — CHANGES_REQUESTED** [review]

File: src/main/java/de/chojo/repbot/ReputationBot.java (hunk 1 above)
Comment (DRSchlaubi) [reply src/main/java/de/chojo/repbot/ReputationBot.java:120]: that's what @RainbowDashLabs said

**Review by dlsf — APPROVED** [review]

**Pull request merged**